unit/util/simplify_expr.cpp

```diff
@@ -145,3 +145,159 @@ TEST_CASE("Simplify Java short -> int -> short casts")
 {
   test_unnecessary_cast(java_short_type());
 }
+
+SCENARIO("Comparing pointers to struct and union members")
+{
+  config.set_arch("none");
+  symbol_tablet symbol_table;
+  namespacet ns(symbol_table);
+
+  signedbv_typet int_type(32);
+
+  struct_typet simple_struct;
```

Comment:
For readability, consider adding a comment showing in C like syntax what you are building:
```C
struct { int x; int y; } s;
union {int x; int y; } u;
```

Reply:
Oh right, I meant to do that but it got lost in refactoring. I will rectify.